File: python/2개이하로다른비트.py
def solution(numbers):
    answer = []
    for n in numbers:
        binary = format(n, 'b')
        count = -1
        idx = len(binary) - 1
        while idx > -1 and binary[idx] != '0':
            idx -= 1
            count += 1
        if count > -1:
            answer.append(n + 2 ** count)
        else:
            answer.append(n + 1)
    return answer

File: JavaScript/2개이하로다른비트.js
function solution(numbers) {
    const answer = [];
    numbers.forEach((n) => {
        const binary = n.toString(2);
        let idx = binary.length - 1, count = -1;
        while (idx >= 0 && binary[idx] !== '0') {
            idx -= 1;
            count += 1;
        }
        answer.push(count > -1 ? n + 2 ** count : n + 1);
    });
    return answer;
}
